### DISCUSSION TITLE
Release 1.2.0 Beta 9

### DISCUSSION THREAD

**Discussion by apohl79** [apohl79/audiogridder#946] [General]
## PLUGIN
- **NEW** Added option to the instrument plugin to transfer audio only when MIDI notes are playing (resolves #941)
- Improved instruments audio streaming (should fix some issues in logic that have been reported)
- Reorganized and cleaned up plugin menus
- Set the AAX category properly for the FX and Instruments plugin
- Improved category submenus in the plugin search
- Minor fixes

## SERVER
- Try more I/O layouts when loading a plugin to enable as many channels as possible
- Deactivated MJPEG video encoder
- Fixed a plugin sandbox load issue: Handle plugins that report infinity for the tail seconds.
- Fixed sandbox deletion on shutdown that could crash the server when rescanning
- Fixed mouse position calculation on windows with scaling enabled
- Fixed port check on macOS
- Minor fixes

<hr /><em>This discussion was created from the release <a href='https://github.com/apohl79/audiogridder/releases/tag/release_1_2_0_beta9'>Release 1.2.0 Beta 9</a>.</em>

**Comment by manatactical** [comment]
Awesome! Gotta try this update tomorrow.

**Comment by JamesXXV** [comment]
I just tried it and yes it does fix the logic crackling issue, but interesting thing happens when I try to close the UI of virtually any plugin, both in FX and instrument versions, as soon as I close the UI it stutters and becomes unresponsive for almost 2-3 seconds before working as usual and if the track is playing it will stutter and sometimes throw a system overload message on logic, also there are sudden CPU spikes in the midst of playing an instrument, isn't easily reproduced on the FX version. Now as I write this bug report I also want to thank you for creating this invaluable tool it has helped me a lot in distributing the sometimes enormous production load between computers so yeah, thanks a lot.

edit: the plugin itself doesn't crash internally it just stutters everything.

**Reply by apohl79** [reply]
thanks for reporting this, I will have a closer look and create a bug issue.

> Now as I write this bug report I also want to thank you for creating this invaluable tool it has helped me a lot in distributing the sometimes enormous production load between computers so yeah, thanks a lot.

cheers, glad it helps you and your workflow!

**Reply by apohl79** [reply]
>when I try to close the UI of virtually any plugin, both in FX and instrument versions, as soon as I close the UI it stutters and becomes unresponsive for almost 2-3 seconds before working as usual

has this started in beta9 or did you see this happen before?

**Reply by apohl79** [reply]
are you using sandboxing and if so, which mode?

**Reply by JamesXXV** [reply]
I am testing it right now, I was going to test builds beta7-9 yesterday but alas It was too late, I'll have the test results in an hour or so.

**Reply by JamesXXV** [reply]
Client: Logic Pro 10.7.3, M1 Pro, Mac OS 12.2.1
Server: Win 10 Pro For Workstations, I5-7400, VMware ESXi

For these tests I version matched the client plugins and the server application.

Beta 6: everything works fine, occasional crackles(quite rare sometimes)

Beta 7: Logic buffer issue exists crackling all instruments and system overloads, some plugins don’t load, issue not seen in version 6

Beta 8: Logic buffer issue exists similarly crackling all instruments and system overloads, some plugins don’t load, issue not seen in version 6

Beta 9: Chain isolation audio glitch/stutter when closing the plugin UI , same for disabled mode and disabled local mode, in Plugin isolation everything works fine, plugin loading issue in beta7 and 8 is fixed too, so is the logic crackling issue

I know the beta 7-8 issues are irrelevant since you've fixed them I just wanted to create a timeline as to what issues existed when, beta9 seems to work as good as beta 6 in plugin isolation mode, at least on Logic Pro

**Reply by apohl79** [reply]
awesome, thanks!

**Reply by JamesXXV** [reply]
found one more, in beta 9 if I keep the UI open while playing an instrument it crackles every now then (single core overload)

**Reply by apohl79** [reply]
What are your AG buffer settings when you get the audio dropout with beta9?

**Reply by JamesXXV** [reply]
My AG buffer settings are 2 blocks (+43 ms), Process buffer range : Medium , I/O buffer size : 256 samples, this used to work perfectly with beta 6.

**Reply by JamesXXV** [reply]
it is with sadness I report that beta 9 server sometimes randomly restarts or worse gets stuck in a restart loop, currently I am on beta 6 again (because I gotta work too) but if you want I'll once again load beta 9 and send you the crash logs (just tell me how to)

**Reply by apohl79** [reply]
cheers, if you had crash reporting enabled, i should find those crashes.

**Reply by JamesXXV** [reply]
I didn't have that enabled, I should do it within a day or two tho, you'll see the logs then

**Reply by JamesXXV** [reply]
well I don't think you have seen the crash logs cause I couldn't reproduce it, maybe an installation specific thing but unfortunately now the beta6 is crackling ever minute or so (after downgrading) I rechecked the settings and I had to move my buffer to 512 for a decent experience, idk how this happened but I even checked the server side settings I cannot find a cause of this sudden crackling.

**Reply by JamesXXV** [reply]
Project with 9 AG instances 0 buffer 512 samples (Rosetta)
<img width="285" alt="Screen Shot 2022-05-27 at 2 55 54 PM" src="https://user-images.githubusercontent.com/94138763/170672077-429036b1-a98b-49d4-8769-bafc94d01071.png">

Same Project with 2 blocks buffer 512 samples (Rosetta)
<img width="274" alt="Screen Shot 2022-05-27 at 2 56 33 PM" src="https://user-images.githubusercontent.com/94138763/170672280-cb1c9afd-99be-477a-a445-2e349caf51c8.png">

for some reason if I create a project in rosetta mode the AG plugin grays out if I switch to native mode
but the vice versa is not true, all native projects work fine in rosetta mode 
(the plugin seems to have better performance too) 

EDIT: also see the last core doing a lot of the heavy lifting in the second picture, that is what sometimes overloads and causes crackling.

**Comment by Pete3030** [comment]
I am on OS High Sierra, DAW is Logic: With Beta9 I have to manually reload the plugins, otherwise connection status would stay red. With Beta7 the loading of the plugins runs smoothly. Reverted back to Beta7..for now ;)

**Reply by apohl79** [reply]
When it stays red, it does not connect. Does your server IP change? This will be more robust going forward.

**Reply by Pete3030** [reply]
My server has a static IP adress, would you suggest to change that to a dynamic IP? I thought AG needs a static IP to run correctly. Anyway, Beta7 runs smooth with a static IP, so something must have been changed with Beta9 for the bad..

**Reply by apohl79** [reply]
No static is good. If you were not able to connect at all, I guess you were running different versions of the server and plugin. Otherwise it should be able to connect and might run into other errors.

**Reply by Pete3030** [reply]
Well, I can get it connecting by clicking reload in the plugin which then disconnects the server and reconnects again, after that the Plugin gets loaded and will be displayed in the Plugin window (The plugin name it was greyed out before, so I could not open it). With Beta7 the connection between the Server and the plugin seems to establish easier right from the start, so less greyed out plugin names, less need to reload manually. Hope this clarifies my issue with Beta9 a little better..

**Reply by apohl79** [reply]
ok thx for this. I think this will be fixed in the next version.

**Reply by Pete3030** [reply]
Ok, that's great to hear! Will try Beta10 as soon as it's available then.. Thx!

**Comment by rasta37Q** [comment]
how do i get the new beta to try

**Reply by apohl79** [reply]
https://audiogridder.com/download/
